### DISCUSSION TITLE
oninstance var

### DISCUSSION THREAD

**Discussion by LoneDespair** [godotengine/godot-proposals#4323] [Scripting]
The onready var is a great and safe way of getting references, but does have a bit of a problem

Most scripts are initialized at _ready function, but data/overwrites that needs to be distributed on a scene for setup, is only propagated after ready

So why not just call the setup function again after propagating your data?
- This is a great approach, however it would mean that setup functions which are already called at _ready would be called again, which is not always ideal

So just stop putting your setup function on _ready func
- Subscenes that depend on calling a setup function, could no longer run on their own

There are plenty of workarounds, but they all have a some drawbacks

**Comment by dalexeev** [comment]
* See #5718.

You opened this discussion first, but mine describes the problem and the solution in more detail, plus it got more response. So I'm closing this discussion. If you want to add something, please write in #5718.
